### ISSUE TITLE
Waiting sound is heard before call interface is shown

### ISSUE THREAD

**Issue by Antreesy** [nextcloud/spreed#13978]
### How to use GitHub

* Please use the 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to show that you are affected by the same issue.
* Please don't comment if you have no relevant information to add. It's just extra noise for everyone subscribed to this issue.
* Subscribe to receive notifications on status change and new comments.

---

## Steps to reproduce
1. Start a call with a lot of participants / imitate server load
2. Hear waiting sound
https://github.com/nextcloud/spreed/blob/32ba79dec837d3c2f33015ccb2522bc4b4534b46/src/utils/webrtc/webrtc.js#L271
4. See call interface later

Idea is to cause signaling/webrtc to register client as 'In call', thereas interface is not ready yet. 
Introduced by https://github.com/nextcloud/spreed/pull/13329 : client now waits for two events: 
- Join the call (signaling-join-call)
- Wait for the users list (signaling-users-changed) EXTERNAL server event

### Expected behaviour
No sound until interface is shown

### Actual behaviour
What title says

## Talk app

**Talk app version:** 20.1.0

**Custom Signaling server configured:** yes

**Comment by nickvergessen** [comment]
> ## Expected behaviour
> No sound until interface is shown

I disagree, expected behavior is that the call UI is loaded quicker, not to delay the call sound. The start call request in our call takes 10+ seconds the last time i checked.
I don't think the user should be left idling like this until all the backend pushing is done.

**Comment by DorraJaouad** [comment]
Yet, we can't know if it will be successful until both events are received . We don't let user pass into call view when there is an error

**Comment by DorraJaouad** [comment]
> In all 3 cases you will never receive that signaling message

So we can revert waiting for both signaling messages (join and users list) if we receive users list first (the signaling message that triggers the sound). 
Otherwise, if the order is normal (join signaling -> users list signaling), we wait for both. 

Is it okay?

**Comment by nickvergessen** [comment]
We solved this, right?

**Comment by Antreesy** [comment]
Fixed with #18571
